### PR TITLE
Cobbler 3.0 fixes

### PR DESCRIPTION
**Pull request by mateiw**
## What does this PR change?

Fixes some `ClassCastExceptions` caused by changes in the Cobbler XML-RPC return types.

## GUI diff

No difference.

## Documentation
- No documentation needed

## Test coverage
- No tests: already covered by cucumber tests

## Links

Fixes https://github.com/SUSE/spacewalk/issues/7149

### CODE REVIEW

**Comment by mcalmer** [comment]
We need to define how it should be. I think the API change is a bug. So we may need investment in cobbler instead of java.

File: java/code/src/com/redhat/rhn/manager/kickstart/cobbler/CobblerLoginCommand.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import org.apache.log4j.Logger;
 
 import java.util.ArrayList;
+import java.util.Base64;
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/manager/kickstart/cobbler/CobblerLoginCommand.java]: Where is this used?

File: java/code/src/com/redhat/rhn/manager/kickstart/cobbler/CobblerLoginCommand.java [hunk 2]
```diff
@@ -49,7 +50,8 @@ public String login(String usernameIn, String passwordIn) {
         args.add(passwordIn);
         String retval = null;
         try {
-            retval = (String) helper.invokeMethod("login", args);
+            byte[] retvalBytes = (byte[]) helper.invokeMethod("login", args);
+            retval = new String(retvalBytes);
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/manager/kickstart/cobbler/CobblerLoginCommand.java]: I made a change in cobbler to change the return value from bytes to string. Maybe this was not enough.

File: java/code/src/com/redhat/rhn/taskomatic/task/CobblerSyncTask.java [hunk 3]
```diff
@@ -68,12 +68,16 @@ public void execute(JobExecutionContext ctxIn)
 
             Double mtime = null;
             try {
-                mtime = (Double) invoker.invokeMethod("last_modified_time",
+                String mtimeStr = (String) invoker.invokeMethod("last_modified_time",
                         new ArrayList());
+                mtime = Double.parseDouble(mtimeStr);
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/taskomatic/task/CobblerSyncTask.java]: I cannot think that these API changes are wanted.

**Comment by mateiw** [comment]
gitarro no changelog needed !!!

**Comment by mcalmer** [comment]
I just learned that there are really some wanted API changes between cobbler 2.6 an 3.0.
So the challenge is now to find out which are wanted API changes and which not.

In case this is wanted we need to change the java code and in the other case we need to change the cobbler code.

**Comment by mateiw** [comment]
Rolled back the changes in the login.

**Pull request merged**